admin-bulk-user-membership/tool/src/main/java/org/sakaiproject/bulk/membership/service/BulkUserMembershipToolService.java

```diff
@@ -0,0 +1,108 @@
+package org.sakaiproject.bulk.membership.service;
+
+import java.util.List;
+import java.util.Collection;
+import java.util.Locale;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+import org.sakaiproject.authz.api.Member;
+import org.sakaiproject.bulk.membership.exception.UsersByEmailException;
+import org.sakaiproject.exception.IdUnusedException;
+import org.sakaiproject.site.api.SiteService;
+import org.sakaiproject.site.api.Site;
+import org.sakaiproject.tool.api.SessionManager;
+import org.sakaiproject.user.api.PreferencesService;
+
+import org.sakaiproject.user.api.User;
+import org.sakaiproject.user.api.UserDirectoryService;
+import org.sakaiproject.user.api.UserNotDefinedException;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class BulkUserMembershipToolService {
+
+    @Autowired
+    private UserDirectoryService userDirectoryService;
+
+    @Autowired
+    private SessionManager sessionManager;
+
+    @Autowired
+	private PreferencesService preferencesService;
+
+    @Autowired
+    private SiteService siteService;
+
+    public User getUser(String criteria) throws UsersByEmailException {
+        try {
+            User user = userDirectoryService.getUserByEid(criteria);
+            if (user != null) {
+                return user;
+            }
+        } catch (UserNotDefinedException usEx) {
+            Collection<User> users = userDirectoryService.findUsersByEmail(criteria);
+            if (users.size() > 1) {
+                throw new UsersByEmailException(criteria);
+            } else if (!users.isEmpty()) {
+                return users.toArray(new User[users.size()])[0];
+            }
+        }
+        return null;
+    }
+
+    public Site getSite(String id) {
+        if (siteService.siteExists(id)) {
```

Comment:
This makes two requests,

            try {
                return siteService.getSite(id);
            } catch (IdUnusedException idEx) {
                return null;
            }